### PR TITLE
fix races reported by rv-predict

### PR DESCRIPTION
**Pull request by dwightguth**
**do not review yet**

### CODE REVIEW

**Comment by dwightguth** [comment]
Jenkins: test this please

**Comment by dwightguth** [comment]
So I really have no idea what the hell is happening here. NoClassDefFoundError? @yilongli if you have any idea you are welcome to suggest otherwise I am probably inclined not to bother with the Java Backend parts of this diff because honestly it's not worth the time spent figuring what the hell happened.

**Comment by yilongli** [comment]
@dwightguth Can we hold on here please? I would like to send in the patch for the data race myself once I am absolutely sure it's not a false positive. Right now, you can best help me by commenting on the issue telling me whether it looks like a false positive. Thanks!

File: java-backend/src/main/java/org/kframework/backend/java/kil/Variable.java [hunk 1]
```diff
@@ -161,11 +161,13 @@ public ASTNode accept(Transformer transformer) {
     Object readResolve() {
         if (anonymous) {
             int id = Integer.parseInt(name.substring(VARIABLE_PREFIX.length()));
-            if (id < counter) {
-                return deserializationAnonymousVariableMap.get(Pair.of(id, sort), this::getFreshCopy);
-            } else {
-                counter = id + 1;
-                return this;
+            synchronized (Variable.class) {
```
Comment (yilongli) [inline java-backend/src/main/java/org/kframework/backend/java/kil/Variable.java:164]: Which race are you trying to fix here?

Comment (dwightguth) [inline java-backend/src/main/java/org/kframework/backend/java/kil/Variable.java:164]: There was a race reported on counter. But I am inclined to suspect an atomic integer operation is probably better here too.

Comment (yilongli) [inline java-backend/src/main/java/org/kframework/backend/java/kil/Variable.java:164]: Can you file an issue with the race report attached? I would like to take a look. Besides, we want to keep track of the real races found by RV-Predict.

**Comment by dwightguth** [comment]
Closing. Yilong can fix these when he has time and get my help if he is not familiar enough with the code.

**Pull request closed without merging**